scripts/shared/dbRequestUtils.js

```diff
@@ -0,0 +1,201 @@
+ * @param {PostRequestOptions} options.postOptions - If a POST request is used,
+ *                                                   contains the specifics of
+ *                                                   the request.
+ * @return {Promise} - A promise that resolves the configured step.
+ */
+gpii.dbRequest.configureStep = function (details, options) {
+    var togo = fluid.promise();
+    var response = gpii.dbRequest.createResponseHandler(
+        details.responseDataHandler,
+        options,
+        togo,
+        details.responseErrMsg
+    );
+    var request;
+    if (details.dataToPost) {
+        request = gpii.dbRequest.createPostRequest(
+            details.dataToPost, response, options, togo
+        );
+    } else {
+        request = gpii.dbRequest.queryDatabase(
+            details.requestUrl, response, details.requestErrMsg, togo
+        );
+    }
+    request.end();
+    return togo;
+};
+
+/**
+ * Create an http request for a bulk docs POST request using the given data.
+ * @param {Object} dataToPost - JSON data to POST and process in bulk.
+ * @param {ResponseCallback} responseHandler - http response callback for the
+ *                                             request.
+ * @param {Object} options - Post request options:
+ * @param {PostRequestOptions} options.postOptions - the POST request specifics.
+ * @param {Promise} promise - promise to reject on a request error.
+ * @return {http.ClientRequest} - An http request object.
+ */
+gpii.dbRequest.createPostRequest = function (dataToPost, responseHandler, options, promise) {
+    var batchPostData = JSON.stringify({"docs": dataToPost});
+    options.postOptions.headers["Content-Length"] = Buffer.byteLength(batchPostData);
+    var batchDocsRequest = http.request(options.postOptions, responseHandler);
+    batchDocsRequest.on("error", function (e) {
+        fluid.log("Error with bulk 'docs' POST request: " + e);
+        promise.reject(e);
+    });
+    batchDocsRequest.write(batchPostData);
+    return batchDocsRequest;
+};
+
+/**
+ * Generate a response handler, setting up the given promise to resolve/reject
+ * at the correct time.
+ * @param {ResponseDataHandler} handleEnd - Function that processes the response
+ *                                          data when the response receives an
+ *                                          "end" event.
+ * @param {Object} options - Data loader options passed to `handleEnd()`.
+ * @param {Promise} promise - Promise to resolve/reject on a response "end" or
+ *                           "error" event.
+ * @param {String} errorMsg - Optional error message to prepend to the error
+ *                            received from a response "error" event.
+ * @return {ResponseCallback} - Reponse callback function suitable for an http
+ *                              request.
+ */
+gpii.dbRequest.createResponseHandler = function (handleEnd, options, promise, errorMsg) {
+    if (!errorMsg) {
```

Comment:
These 3 lines can be simplified to `errorMsg = errorMsg || "";`

Reply:
Ok